app/upgrades/v16/concentrated_pool.go

```diff
@@ -0,0 +1,154 @@
+package v16
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	"github.com/osmosis-labs/osmosis/v15/app/keepers"
+	clmodel "github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/model"
+	gammkeeper "github.com/osmosis-labs/osmosis/v15/x/gamm/keeper"
+	gammtypes "github.com/osmosis-labs/osmosis/v15/x/gamm/types"
+	incentiveskeeper "github.com/osmosis-labs/osmosis/v15/x/incentives/keeper"
+	incentivestypes "github.com/osmosis-labs/osmosis/v15/x/incentives/types"
+	poolincentiveskeeper "github.com/osmosis-labs/osmosis/v15/x/pool-incentives/keeper"
+	"github.com/osmosis-labs/osmosis/v15/x/poolmanager"
+	poolmanagertypes "github.com/osmosis-labs/osmosis/v15/x/poolmanager/types"
+)
+
+// getGaugesForCFMMPool returns the gauges associated with the given CFMM pool ID, by first retrieving
+// the lockable durations for the pool, then using them to query the pool incentives keeper for the
+// gauge IDs associated with each duration, and finally using the incentives keeper to retrieve the
+// actual gauges from the retrieved gauge IDs.
+func getGaugesForCFMMPool(ctx sdk.Context, incentivesKeeper incentiveskeeper.Keeper, poolIncentivesKeeper poolincentiveskeeper.Keeper, poolId uint64) ([]incentivestypes.Gauge, error) {
+	lockableDurations := poolIncentivesKeeper.GetLockableDurations(ctx)
+	cfmmGaugeIds := make([]uint64, 0, len(lockableDurations))
+	for _, duration := range lockableDurations {
+		gaugeId, err := poolIncentivesKeeper.GetPoolGaugeId(ctx, poolId, duration)
+		if err != nil {
+			return nil, err
+		}
+		cfmmGaugeIds = append(cfmmGaugeIds, gaugeId)
+	}
+
+	cfmmGauges, err := incentivesKeeper.GetGaugeFromIDs(ctx, cfmmGaugeIds)
+	if err != nil {
+		return nil, err
+	}
+
+	return cfmmGauges, nil
+}
```

Comment:
should we just have this function in poolincentives? Seems like a useful function